internal/restic/snapshot.go

```diff
@@ -27,17 +27,45 @@ type Snapshot struct {
 	id *ID // plaintext ID, used during restore
 }
 
+func pathSplit(path string) (root string, lst []string) {
```

Comment:
This function needs tests, I'll add a test scaffold for you.

Reply:
Okay.